### PR TITLE
Rework "branch" option usage

### PR DESCRIPTION
**Pull request by postatum**
TLDR: Moved an option closer to where it's used.

Option "branch" is a legacy of raml-tck and raml-tck-runner. It makes sense in these projects because each of raml-tck-runner runners clones raml-tck repo/branch, runs tests against it and
stores branch info in JSON report (a step between testing and HTML reports generation) so HTML report generator can generate links to a proper branch of tck github repo.

But this option doesn't make sense in asyncapi/tck since tck tests and the runner are located in the same repo. Thus I've reworked the option.

Now test runners don't accept it and don't store it in JSON reports. Instead Makefile now accepts a non-mandatody option "tck_branch" which is passed directly to HTML reports generator. When not provided its option value is "master", meaning HTML reporter will generate links to tck files in "master" branch of asyncapi/tck.

Also "publish-gh-pages" Github Actions Workflow can be adjusted to call "make tck_branch=somebranch" instead of simply "make". But I think it's not currently necessary because the only branch we run tests against is "master", which is a default when the option is not provided.

### CODE REVIEW

**Comment by fmvilas** [comment]
Feel free to merge it if you think it's ready.
